cmd/sync.go

```diff
@@ -74,36 +71,22 @@ func (raw *rawSyncCmdArgs) parsePatterns(pattern string) (cookedPatterns []strin
 	return
 }
 
-// given a valid URL, parse out the SAS portion
-func (raw *rawSyncCmdArgs) separateSasFromURL(rawURL string) (cleanURL string, sas string) {
-	fromUrl, _ := url.Parse(rawURL)
-
-	// TODO add support for other service URLs
-	blobParts := azblob.NewBlobURLParts(*fromUrl)
-	sas = blobParts.SAS.Encode()
-
-	// get clean URL without SAS
-	blobParts.SAS = azblob.SASQueryParameters{}
-	bUrl := blobParts.URL()
-	cleanURL = bUrl.String()
-
-	return
-}
-
 // validates and transform raw input into cooked input
 func (raw *rawSyncCmdArgs) cook() (cookedSyncCmdArgs, error) {
 	cooked := cookedSyncCmdArgs{}
 
+	// this if statement ladder remains instead of being separated to help determine valid combinations for sync
+	// consider making a map of valid source/dest combos and consolidating this to generic source/dest setups, akin to the lower if statement
 	cooked.fromTo = inferFromTo(raw.src, raw.dst)
 	if cooked.fromTo == common.EFromTo.Unknown() {
 		return cooked, fmt.Errorf("Unable to infer the source '%s' / destination '%s'. ", raw.src, raw.dst)
 	} else if cooked.fromTo == common.EFromTo.LocalBlob() {
-		cooked.destination, cooked.destinationSAS = raw.separateSasFromURL(raw.dst)
+		cooked.destination, cooked.destinationSAS, _ = SplitAuthTokenFromResource(raw.dst, cooked.fromTo.To())
```

Comment:
The usual Go pattern, for cases like this where you intend to ignore the error, would be to write a method like this, and then call it from these places:

```
func MustSplitAuthTokenFromResource(resource string, location Location) (resource, sas, string) {
   resource, sas, err := SplitAuthTokenFromResource(resource, location)
    common.PanicIfErr(err)
}
```

For an example, see regex.Compile vs regex.MustCompile

I think we should do something similar here, given that its easily done, and ignored errors can lead to strange mysteries at support time.